### PR TITLE
Adding a version comparison to the k8s CI tests

### PR DESCRIPTION
**Pull request by mattfarina**
The bot that merges PRs re-runs the e2e job before merging a pull
request. By also having it here (circleci provides fast response
to PR authors) we make sure the version has been incremented in
the chart prior to it being merged.

### CODE REVIEW

File: test/changed.sh [hunk 1]
```diff
@@ -39,6 +39,56 @@ function cleanup {
 }
 trap cleanup EXIT
 
+# Verify that the semver for the chart was increased
+# This function is also in the circleci test runs. The duplication is to make
+# sure this is checked a second time right before merging the PR.
+semvercompare() {
```
Comment (viglesiasce) [inline test/changed.sh]: Can we put these functions in a common file and then source and then source them for both circle and PR tests?

**Review by viglesiasce — CHANGES_REQUESTED** [review]

File: test/changed.sh (hunk 1 above)
Comment (mattfarina) [reply test/changed.sh]: Yeah. It's WIP and I want to get it working first.

**Comment by mattfarina** [comment]
@viglesiasce This is now ready.

**Review by viglesiasce — APPROVED** [review]

**Pull request merged**